server/lib/src/valueset/image.rs

```diff
@@ -0,0 +1,466 @@
+#![allow(dead_code)]
+use std::fmt::Display;
+
+use hashbrown::HashSet;
+use image::codecs::gif::GifDecoder;
+use image::codecs::jpeg::JpegDecoder;
+use image::codecs::webp::WebPDecoder;
+use image::ImageDecoder;
+use kanidm_proto::internal::{ImageType, ImageValue};
+
+use crate::prelude::*;
+use crate::repl::proto::ReplAttrV1;
+use crate::schema::SchemaAttribute;
+use crate::valueset::{DbValueSetV2, ValueSet};
+
+#[derive(Debug, Clone)]
+pub struct ValueSetImage {
+    set: HashSet<ImageValue>,
+}
+
+pub(crate) const MAX_IMAGE_HEIGHT: u32 = 1024;
+pub(crate) const MAX_IMAGE_WIDTH: u32 = 1024;
+/// 128kb should be enough for anyone... right? :D
+pub(crate) const MAX_FILE_SIZE: u32 = 1024 * 128;
+
+pub(crate) trait ImageValueThings {
+    fn validate_image(&self) -> Result<(), ImageValidationError>;
+    fn validate_is_png(&self) -> Result<(), ImageValidationError>;
+    fn validate_is_gif(&self) -> Result<(), ImageValidationError>;
+    fn validate_is_jpg(&self) -> Result<(), ImageValidationError>;
+    fn validate_is_webp(&self) -> Result<(), ImageValidationError>;
+    fn validate_is_svg(&self) -> Result<(), ImageValidationError>;
+
+    /// A sha256 of the filename/type/contents
+    fn hash_imagevalue(&self) -> String;
+
+    fn get_limits(&self) -> image::io::Limits {
+        let mut limits = image::io::Limits::default();
```

Comment:
We likeuly need a plan on how to have these configurable in future.

Reply:
Yeah, I'd say it'll be yet another replicable system setting of some kind.

Reply:
In that case it should be in the fn validate of the valueset because I think we have access to the query server at that point?

Reply:
Or we can make the limits come from the schema for example. I dunno. Thought needed. For now leave it like this.

Reply:
Yeah, as a first-run this'll work :D